libraries/chain/steem_evaluator.cpp

```diff
@@ -1205,10 +1235,26 @@ namespace golos { namespace chain {
                 const auto& comment_vote_idx = _db.get_index<comment_vote_index>().indices().get<by_comment_voter>();
                 auto itr = comment_vote_idx.find(std::make_tuple(comment.id, voter.id));
 
-                int64_t elapsed_seconds = (_db.head_block_time() - voter.last_vote_time).to_seconds();
+                auto elapsed_seconds = (_db.head_block_time() - voter.last_vote_time).to_seconds();
+
+                if (_db.has_hardfork(STEEMIT_HARDFORK_0_19__533)) {
+                    auto consumption = mprops.comments_window / mprops.comments_per_window;
+
+                    auto regenerated_capacity = std::min(mprops.votes_window, uint16_t(elapsed_seconds));
+                    auto current_capacity = std::min(uint16_t(voter.voting_capacity + regenerated_capacity), mprops.comments_window);
+
+                    GOLOS_CHECK_BANDWIDTH(current_capacity, consumption,
+                        bandwidth_exception::vote_bandwidth,
+                        "Can only vote ${vpw} times in ${vw} seconds.",
+                        ("cpw", mprops.votes_per_window)("cw", mprops.votes_window));
```

Comment:
Give items normal names, they are used by clients to detect errors.

You should figure out how the errors subsystem works.